src/Core/Framework/Event/EventData/EntityType.php

```diff
@@ -2,17 +2,26 @@
 
 namespace Shopware\Core\Framework\Event\EventData;
 
+use Shopware\Core\Framework\DataAbstractionLayer\EntityDefinition;
+
 class EntityType implements EventDataType
 {
     public const TYPE = 'entity';
 
     /**
-     * @var string
+     * @var class-string<EntityDefinition>
      */
-    private $definitionClass;
+    private string $definitionClass;
 
-    public function __construct(string $definitionClass)
+    /**
+     * @param class-string<EntityDefinition>|EntityDefinition $definitionClass
+     */
+    public function __construct(string|EntityDefinition $definitionClass)
```

Comment:
As we are now able to use union types i like the change! 

prior to php 8 i would have disliked it, as we would have to remove the native typing completely, but in this way the change totally makes sense